syil_lnc_toolpath.cps

```diff
@@ -1673,23 +1675,49 @@ function onCyclePoint(x, y, z) {
       writeBlock(gFormat.format(65), '"PROTECTEDMOVE"', zOutput.format(z - cycle.depth));
 
       WCS_CODE = getProbingArguments(cycle, probeWorkOffsetCode);
+    
+      if(cycle.approach1== "positive")
+      {
+      EXPECTED_Y = yOutput.format(y + cycle.probeClearance + tool.diameter / 2);
+      EXPECTED_X = xOutput.format(x + cycle.probeClearance + tool.diameter / 2);
+      }
+      else
+      {
+      EXPECTED_Y = yOutput.format(y - cycle.probeClearance - tool.diameter / 2);
+      EXPECTED_X = xOutput.format(x - cycle.probeClearance - tool.diameter / 2);
+      }
+
+      EXPECTED_Z = "Z"+xyzFormat.format(cycle.stock - cycle.depth);
 
       DISTANCE = approach(cycle.approach1)*(cycle.probeClearance + tool.diameter / 2 + cycle.probeOvertravel)
       B_ARG = "B"+xyzFormat.format(DISTANCE)
 
-      writeBlock(gFormat.format(65), '"PROBEX"', WCS_CODE, B_ARG);
+      writeBlock(gFormat.format(65), '"PROBEX"', WCS_CODE[7], WCS_CODE[8], B_ARG);
+      writeBlock(gFormat.format(65), '"CHECKPOSITIONALTOLERANCE"', WCS_CODE[8], WCS_CODE[9], WCS_CODE[3],'V1', EXPECTED_X, EXPECTED_Y, EXPECTED_Z);
       break;
     case "probing-y":
       forceXYZ();
       // writeBlock(gFormat.format(31), "P2 ", zOutput.format(z - cycle.depth), "F50");  // protected positioning move 
       writeBlock(gFormat.format(65), '"PROTECTEDMOVE"', zOutput.format(z - cycle.depth));
 
       WCS_CODE = getProbingArguments(cycle, probeWorkOffsetCode);
+      if(cycle.approach1== "positive")
```

Comment:
check out the `approach` function. That can be used to simplify this code a bit.